pkg/net/ip.go

```diff
@@ -242,3 +244,62 @@ func IPString(ip cue.Value) (string, error) {
 	}
 	return ipdata.String(), nil
 }
+
+func netIPAddOffset(addr netip.Addr, offset *big.Int) (netip.Addr, error) {
+	i := big.NewInt(0).SetBytes(addr.AsSlice())
+	i = i.Add(i, offset)
+
+	if i.Sign() < 0 {
+		return netip.Addr{}, errors.New("underflowed")
```

Comment:
Maybe be a bit clearer with the error?

```suggestion
		return netip.Addr{}, errors.New("IP address arithmetic resulted in out-of-range address (underflow)")
```
(similarly below)